### ISSUE TITLE
Kucoin - Timeout error for multiple market pairs

### ISSUE THREAD

**Issue by RC-13** [CoinAlpha/hummingbot#3230]
**Describe the bug**
Random time out for a bunch of market pairs keeps coming up when using KuCoin connector on the development branch. Despite errors coming up, it doesn't affect existing and creation of orders. Also, this timeout adds up to the memory leak issue seen on Kucoin (#3223)
```python
2021-04-21 16:45:16,102 - 7249 - hummingbot.connector.exchange.kucoin.kucoin_api_order_book_data_source - WARNING - Message recv() timed out. Stream type = StreamType.Depth,Trading pairs = {'PROPS-USDT', 'KLV-USDT', 'FIL-USDT', 'FRM-USDT', 'ORAI-USDT', 'ZEN-USDT', 'PRE-USDT', 'BOA-USDT', 'MITX-USDT', 'AAVE-UST', 'GRT-USDT', 'WIS-BTC', 'TRIAS-USDT', 'COMB-USDT', 'LINK-USDC', 'MIR-UST', 'REN-USDT', 'UNFI-USDT', 'DOGE-USDT', 'BCHA-USDT', 'DAO-USDT', 'CAS-USDT', 'CRV-USDT', 'TRIAS-BTC', 'BUX-BTC', 'USDT-UST', 'HTR-USDT', 'CUDOS-USDT', 'ETH2-ETH', 'PLU-USDT', 'API3-USDT', 'DOGE-USDC', 'LRC-USDT', 'YFI-UST', 'OPCT-ETH', 'GOM2-BTC', 'CTI-USDT', 'ETH-UST', 'LTX-BTC', 'LON-USDT', 'GOM2-USDT', 'BRG-USDT', 'REVV-USDT', 'BCHSV-USDC', 'SNX-UST', 'RFUEL-USDT', '1INCH-USDT', 'ATOM-UST', 'UDOO-ETH', 'THETA-USDT', 'BCH-USDC', 'LTC-USDC', 'USDC-USDT', 'DFI-BTC', 'UBX-ETH', 'QNT-USDT', 'PROPS-BTC', 'UQC-ETH', 'LOC-USDT', 'DOT-UST', 'BRG-BTC', 'MSWAP-BTC', 'AAVE-USDT', 'LUNA-UST', 'FRONT-USDT', 'MIR-USDT', 'LTX-USDT', 'UQC-BTC', 'TONE-ETH', 'WBTC-BTC', 'REVV-BTC', 'TONE-BTC', 'VIDT-USDT', 'BTC-UST', 'SHR-USDT', 'SUSHI-USDT', 'XHV-USDT', 'AAVE-BTC', 'USDC-UST', 'UBXT-USDT', 'ADA-USDC', 'MITX-BTC', 'HYDRA-USDT', 'STRONG-USDT', 'DFI-USDT', 'ROSE-USDT', 'CAKE-USDT', 'BAT-USDT', 'SHR-BTC', 'CTI-ETH', 'REAP-USDT', 'MSWAP-USDT', 'FRONT-BTC', 'WBTC-ETH', 'CAS-BTC', 'OPCT-BTC', 'KLV-BTC', 'DOGE-BTC', 'EOS-USDC', 'ZEE-USDT'}.
2021-04-21 16:45:17,859 - 7249 - hummingbot.connector.exchange.kucoin.kucoin_api_order_book_data_source - WARNING - Message recv() timed out. Stream type = StreamType.Depth,Trading pairs = {'WIN-USDT', 'CHR-USDT', 'FKX-ETH', 'DERO-BTC', 'ATOM-ETH', 'OCEAN-ETH', 'BTC-PAX', 'CRO-BTC', 'XMR-BTC', 'MTV-ETH', 'BTT-USDT', 'NIX-BTC', 'ANKR-BTC', 'PIVX-BTC', 'FET-BTC', 'NRG-ETH', 'XTZ-USDT', 'FTM-USDT', 'TRTL-BTC', 'COTI-USDT', 'OCEAN-BTC', 'DX-ETH', 'ALGO-USDT', 'MTV-USDT', 'DERO-USDT', 'TRTL-ETH', 'NOIA-BTC', 'EOSC-USDT', 'CIX100-USDT', 'RIF-BTC', 'FX-ETH', 'ENQ-BTC', 'ATOM-USDT', 'R-USDT', 'TOMO-USDT', 'ADA-BTC', 'SERO-USDT', 'WXT-USDT', 'XEM-BTC', 'NIX-ETH', 'BTC-USDC', 'ATOM-BTC', 'XMR-USDT', 'DAPPT-USDT', 'CHZ-BTC', 'ONE-BTC', 'BLOC-USDT', 'ETN-USDT', 'ALGO-BTC', 'FX-BTC', 'XTZ-BTC', 'CHR-BTC', 'NOIA-USDT', 'MHC-BTC', 'PIVX-ETH', 'ZEC-BTC', 'RBTC-BTC', 'FORESTPLUS-BTC', 'XEM-USDT', 'NIM-BTC', 'NRG-BTC', 'BNB-USDT', 'FKX-BTC', 'CHZ-USDT', 'COTI-BTC', 'DX-BTC', 'ZEC-USDT', 'CSP-ETH', 'FET-ETH', 'LOKI-ETH', 'MHC-USDT', 'ALGO-ETH', 'ATOM-KCS', 'WIN-BTC', 'LOKI-BTC', 'AVA-USDT', 'TOKO-BTC', 'MTV-BTC', 'NIM-ETH', 'BOLT-BTC', 'ENQ-USDT', 'BNB-BTC', 'WXT-BTC', 'ONE-USDT', 'VSYS-USDT', 'TOKO-USDT', 'TRX-KCS', 'RFOX-USDT', 'ARPA-USDT', 'DAPPT-BTC', 'ADA-USDT', 'BOLT-USDT', 'TEL-USDT', 'ETH-BTC', 'MHC-ETH', 'XMR-ETH', 'FORESTPLUS-USDT', 'JAR-BTC', 'JAR-USDT', 'CSP-BTC'}.
2021-04-21 16:45:21,232 - 7249 - hummingbot.connector.exchange.kucoin.kucoin_api_order_book_data_source - ERROR - Timeout error with WebSocket connection. Retrying after 5 seconds...
Traceback (most recent call last):
  File "/Users/clint_g/Documents/Github-Repo/hb-general/hummingbot/connector/exchange/kucoin/kucoin_api_order_book_data_source.py", line 212, in _inner_messages
    yield await ws.recv()
  File "/Users/clint_g/opt/anaconda3/envs/hummingbot/lib/python3.8/site-packages/websockets/protocol.py", line 509, in recv
    await self.ensure_open()
  File "/Users/clint_g/opt/anaconda3/envs/hummingbot/lib/python3.8/site-packages/websockets/protocol.py", line 811, in ensure_open
    await asyncio.shield(self.close_connection_task)
asyncio.exceptions.CancelledError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/clint_g/Documents/Github-Repo/hb-general/hummingbot/connector/exchange/kucoin/kucoin_api_order_book_data_source.py", line 439, in _collect_and_decode_messages_loop
    async for raw_msg in kucoin_msg_iterator:
  File "/Users/clint_g/Documents/Github-Repo/hb-general/hummingbot/connector/exchange/kucoin/kucoin_api_order_book_data_source.py", line 237, in __aiter__
    async for raw_msg in self._inner_messages(ws):
  File "/Users/clint_g/Documents/Github-Repo/hb-general/hummingbot/connector/exchange/kucoin/kucoin_api_order_book_data_source.py", line 212, in _inner_messages
    yield await ws.recv()
  File "/Users/clint_g/opt/anaconda3/envs/hummingbot/lib/python3.8/site-packages/async_timeout/__init__.py", line 55, in __aexit__
    self._do_exit(exc_type)
  File "/Users/clint_g/opt/anaconda3/envs/hummingbot/lib/python3.8/site-packages/async_timeout/__init__.py", line 92, in _do_exit
    raise asyncio.TimeoutError
asyncio.exceptions.TimeoutError
```

**Steps To Reproduce**
// A concise description of the **steps to reproduce the buggy behavior**:

Here are the steps to reproduce the issue (see attachments in the section below):
1. Setup a pure market making strategy or Avellaneda market making on the development branch
2. Start the bot and observe the log pane, randomly, a timeout message will appear after and list down a bunch of trading pairs

**Screenshots**
PMM: 
![image](https://user-images.githubusercontent.com/57189990/115524800-13d44b80-a2c1-11eb-8519-c79aafc7e00f.png)

Avellaneda MM:
![image](https://user-images.githubusercontent.com/57189990/115524823-1b93f000-a2c1-11eb-9015-f19a0ebffd81.png)

KucoinOrderbookMessage component increasing from time to time:
![image](https://user-images.githubusercontent.com/57189990/115526304-77ab4400-a2c2-11eb-83ae-51ddec81d973.png)

**Release version**
0.39.0

**Attachments**
[logs_kucoin-dev-test.log](https://github.com/CoinAlpha/hummingbot/files/6349410/logs_kucoin-dev-test.log)

*WARNING: Do NOT publish any exchange API keys or your wallet's private key. Whoever has access to them may steal your assets!*

**Optional: your discord username:**

**Comment by rxlxrxsx** [comment]
Additional info regarding this issue, when the `Message recv() timed out` is triggered, it makes the order_book not sync with the exchange

https://user-images.githubusercontent.com/64377055/116499309-725f8200-a8de-11eb-9511-f456546e936c.mp4

[logs_kucoin_eth-usdt.log.zip](https://github.com/CoinAlpha/hummingbot/files/6396149/logs_kucoin_eth-usdt.log.zip)

**Comment by RC-13** [comment]
Issue fixed on the latest development branch (v0.39.0)
